pubsub/google/cloud/pubsub/_gax.py

```diff
@@ -58,21 +66,24 @@ def list_topics(self, project, page_size=0, page_token=None):
                            passed, the API will return the first page of
                            topics.
 
-        :rtype: tuple, (list, str)
-        :returns: list of ``Topic`` resource dicts, plus a
-                  "next page token" string:  if not None, indicates that
-                  more topics can be retrieved with another call (pass that
-                  value as ``page_token``).
+        :rtype: :class:`~google.cloud.iterator.Iterator`
+        :returns: Iterator of :class:`~google.cloud.pubsub.topic.Topic`
+                  accessible to the current API.
         """
         if page_token is None:
             page_token = INITIAL_PAGE
         options = CallOptions(page_token=page_token)
         path = 'projects/%s' % (project,)
         page_iter = self._gax_api.list_topics(
             path, page_size=page_size, options=options)
-        topics = [{'name': topic_pb.name} for topic_pb in page_iter.next()]
-        token = page_iter.page_token or None
-        return topics, token
+        page_iter = functools.partial(_recast_page_iterator, page_iter)
+
+        # NOTE: We don't currently have access to the client, so callers
+        #       that want the client, must manually bind the client to the
+        #       iterator instance returned.
+        return Iterator(client=None, path=path,
+                        item_to_value=_item_to_topic,
+                        page_iter=page_iter)
```

Comment:
The topic instances can be used, but only because I cheat in `Client.list_topics` and put back the client

```python
iterator = low_level_api.list_topics(...)
iterator.client = self
return iterator
```

> Wouldn't we be better off changing the point-of-instantiation of the GAX API objects to pass in the client for use here.

Yes, just wanted to keep the commit as focused as I could. Happy to modify the signature in this PR.